stock_picking_invoice_link/models/stock_move.py

```diff
@@ -30,7 +31,14 @@ def write(self, vals):
         if "product_uom_qty" in vals and not self.env.context.get(
             "bypass_stock_move_update_restriction"
         ):
-            for move in self:
+            for move in self.filtered(
```

Comment:
I would do it inside the loop to not iterate twice the recordset (filtered + for).

Reply:
As far as I understand "filtered" function limit records in "for" function iteration, Please can you explain me a bit why iterate twice?